Switch to Zammad 6.3

mgruner opened the pull request:
#### What this PR does / why we need it

Make neccessary changes for Zammad 6.3:

- Remove the mount point for the `var/` folder as it is not used any more by Zammad.
- Add postgresql option parameter for connection pool size (required by new Rails version).
- Improve handling of autowizard secret by mounting it in the FS directly.
- This also improves startup time as the `zammad-init` container is not needed any more, unless there is `customInit` code present.

#### Special Notes for your Reviewer

There is one backwards incompatible change here. If an existing secret is used for the auto wizard, it must not be (double) base64 encoded any more. That's why I went for a minor update, is that ok? I'm not sure where to document this.

#### Checklist

- [x] Chart Version bumped

monotek reviewed (commented):
lgtm :)

Seemst to me like a major version update though, if existing functionality changes and special adjustments for upgrading users might be needed.

So changes should be reflected in upgrading docs too.

mgruner commented:
@monotek thanks for the feedback. I will wait for the 6.3 release then and include it with the changes for that release. I would like to first release a new version with the StatefulSet for it before we address #243 with another major release then, so that there is a fallback option in case things go wrong.

monotek commented on zammad/Chart.yaml:
```diff
@@ -1,7 +1,8 @@
-appVersion: 6.2.0-1
+version: 10.4.0
+# appVersion: 6.3.0-0
+appVersion: develop
```
Do you want to merge it like this?
If so, why?
After zammad/zammad gets the 6.3 tag we also have a real version to release here?

mgruner replied:
No, this MR is only a preperation before the release, which is scheduled in a few weeks. I will merge it only after, and switch the version to the commented out one `6.3.0-0`.

I created this MR so that we can test the changes before, and are ready when the release comes. Sorry for making that not clear enough.

monotek replied:
Ah, ok. Thanks for clarification :)
So beside app version it looks good to me as CI runs successfull.
I'll approve as soon as the correct version is set :)

mgruner commented:
> lgtm :)
> 
> Seemst to me like a major version update though, if existing functionality changes and special adjustments for upgrading users might be needed.
> 
> So changes should be reflected in upgrading docs too.

@monotek could you have another look, please? I changed it to a major version increase, and added upgrading instructions.

mgruner commented:
@monotek could you please have a look again? Zammad 6.3 was released today, and I changed the version to the final one for 6.3.0 now.
